### PR TITLE
[bondhome] Add raw speed channel

### PR DESCRIPTION
**Pull request by ktgeek**
Adds a channel for the raw speed value from the bond api without being interpreted as a dimmer. This is useful for things like 3 speed ceiling fans integrating with the homekit integration that understand smaller ranges and can interact more directly. This is marked for advanced usage as most users will probably be fine with using a dimmer.

JARs built with java 17 against the 3.4 branch: https://github.com/ktgeek/openhab-addons/releases/download/bondhome-20230103-01/org.openhab.binding.bondhome-3.4.1-SNAPSHOT.jar
JAR built against the 4.0 branch: https://github.com/ktgeek/openhab-addons/releases/download/bondhome-20230103-01/org.openhab.binding.bondhome-4.0.0-SNAPSHOT.jar

### CODE REVIEW

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ccutrer) [inline bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:209]: please remove this log line; it's easy to get the same information from the events log (yes, I realize the other channels do it; they should also probably be removed)

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ccutrer) [inline bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:221]: maybe log the action as well as the value, since it might be TURN_OFF?

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ccutrer) [inline bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:645]: this seems buggy. fpfanspeed should either be 1-100, or 1-{max speed}, but not both. while we're at it, is fpfanspeed 0 when the fan is off?

Comment (ccutrer) [reply bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:645]: ohhh fpfanspeed is _fireplace_ fan speed. your handleCommand implementation definitely does not handle a raw fan speed with a fireplace fan

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:221]: it would show up as null there, but i can move that up, I agree it makes more logical sense to talk about off vs speed.

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:645]: I'll look deeper at this... for some reason I thought this would be ok, because speed is speed if you feed it, but  maybe i missed the winkle with the fireplace fan.  Seems like an easy fix via an if statement.

File: bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml
Comment (jlaur) [inline bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml:39]: Perhaps this should be:
```suggestion
	<channel-type id="rawFanSpeedChannelType" advanced="true">
```

File: bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml
Comment (jlaur) [inline bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml:43]: This category/icon doesn't exist, but could be:
```suggestion
		<category>Fan</category>
```

or same as for `fanSpeedChannelType`, if this is actually the same fan, i.e. `Heating`.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Two minor comments from my side.

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:645]: after talking to @ccutrer in another medium, we thought we'd just not set this for fireplace fans for now

File: bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml:39]: Thanks! TIL

File: bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/thing/Channels.xml:43]: I'll switch it to Heating to match the Dimmer interpreted fan speed

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:219]: Went with the "just set it to off" path.

**Comment by ktgeek** [comment]
New jars:

- https://github.com/ktgeek/openhab-addons/releases/download/bondhome-20230105-01/org.openhab.binding.bondhome-3.4.1-SNAPSHOT.jar
- https://github.com/ktgeek/openhab-addons/releases/download/bondhome-20230105-01/org.openhab.binding.bondhome-4.0.0-SNAPSHOT.jar

**Comment by jlaur** [comment]
@ccutrer - I'll let you check again and approve if it looks good to you.

**Comment by ktgeek** [comment]
@ccutrer when you get a chance. I am not wild about the nested if, but it seemed to match our intentions the best

**Review by ccutrer — APPROVED** [review]

File: bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/i18n/bondhome.properties
Comment (jlaur) [inline bundles/org.openhab.binding.bondhome/src/main/resources/OH-INF/i18n/bondhome.properties:73]: Seems like regeneration was missed, in Channels.xml the text is:
```suggestion
channel-type.bondhome.rawFanSpeedChannelType.description = Sets fan speed using raw Bond values
```

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:592]: It's added when ceiling fan, but the comment above says "add (...) unless we're a ceiling fan"?

**Review by jlaur — CHANGES_REQUESTED** [review]
I've added two additional findings.

**Comment by ktgeek** [comment]
oh well, that didn't do what I wanted.  (I really got stop doing this changes when i'm bored in a meeting)

File: bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java
Comment (ktgeek) [reply bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java:592]: I reserved the logic, but forgot to reverse the entire statement around it.  It used to be "unless we're a FP fan."  I'll get that fixed up.

**Review by jlaur — APPROVED** [review]
LGTM - thanks!

**Pull request merged**
